iperf.py

```diff
@@ -84,20 +84,27 @@ def server(self, cmd: str):
         self.exec_thread = ReturnValueThread(target=server, args=(self, cmd))
         self.exec_thread.start()
 
-    def run(self, duration: int):
-        pass
-
-    def stop(self):
+    def stop(self) -> None:
         logger.info(f"Stopping execution on {self.pod_name}")
-        r = self.exec_thread.join()
-        if r.returncode != 0:
-            logger.error(
-                f"Error occured while stopping Iperf server: errcode: {r.returncode} err {r.err}"
-            )
-        logger.debug(f"IperfServer.stop(): {r.out}")
+        self.exec_thread.join()
+        if self.exec_thread.result is not None:
+            r = self.exec_thread.result
+            if r.returncode != 0:
+                logger.error(
+                    f"Error occured while stopping Iperf server: errcode: {r.returncode} err {r.err}"
+                )
+            logger.debug(f"IperfServer.stop(): {r.out}")
+        else:
+            logger.error("Thread did not return a result")
 
-    def output(self, out: common.TftAggregateOutput):
-        pass
+    def run(self, duration: int) -> None:
```

Comment:
We are using the abstract class and polymorphism here. This type of software design makes it such that as long as the base class has an implementation it should be able to call it. The advantage is to make it such that we can lump things together and run them.

Reply:
That makes sense. I reverted those changes removing methods from `IperfServer`.